### ISSUE TITLE
Train YOLO V3 on VisDrone dataset

### ISSUE THREAD

**Issue by MuhammadAsadJaved** [pjreddie/darknet#1811]
I want to train YOLO V3  for object detection on VisDrone dataset which contains images with size about 540 x 960 P etc . and their annotations accordingly,  however  for the YOLO V3 training I want to resize all the images to 416 x 416 and also change Annotations accordingly . 

Is there any method or suggestions to do it ?  
If I just resize all the images to 416 x 416 P and use the original annotations is it ok or I must change the annotations if i change the size?

**Comment by flyskydxx** [comment]
Hi Muhammad,

I think you don't need to change the size of any pic in VisDrone dataset. I've train my yolo weight on original data and it's turn out to be okay.

regards

**Comment by Runkli** [comment]
@flyskydxx Hi there, I was wondering if you could share the pre-trained weights for VisDrone on darknet?

**Comment by MuhammadAsadJaved** [comment]
Hi,
     I am so sorry it's confidential work and we did not train on VisDrone
alone, we combine with other datasets.
This project trained only on VisDrone and with VisDrone, maybe it's what
you are looking for.
Please have a try  https://gitlab.com/EAVISE/jetnet

**Comment by hanifjunos** [comment]
@flyskydxx  how did you train yolo on VisDrone dataset? i can see that the VisDrone text format is different from the text format require to train yolo with Darknet.

**Comment by MuhammadAsadJaved** [comment]
It’s easy to convert annotations. I am out of office till 8 October. I can
help you with that on 9 October.

**Comment by hanifjunos** [comment]
@MuhammadAsadJaved thanks for your help.

**Comment by jhonjam** [comment]
> @flyskydxx how did you train yolo on VisDrone dataset? i can see that the VisDrone text format is different from the text format require to train yolo with Darknet.

hi, i have the same doubt, the visdrone annotations are different from the annotations of yolo (darknet). Can you help me with that problem, or how you solved it. Thank you

**Comment by MuhammadAsadJaved** [comment]
Hi @hanifjunos  @jhonjam ,   You can convert VisDrone annotations to Darknet format by this simple script. 

https://github.com/zhaobaiyu/visdrone#scripts-for-dataset-transform

**Comment by jhonjam** [comment]
> Hi @hanifjunos @jhonjam , You can convert VisDrone annotations to Darknet format by this simple script.
> 
> https://github.com/zhaobaiyu/visdrone#scripts-for-dataset-transform

yes, I am trying to use that code. However, it asks as input in sequence set, which does not have it. those are the downloaded files from the dataset (http://aiskyeye.com/download/object-detection-2/)
![image](https://user-images.githubusercontent.com/62752094/135721170-2f6144b2-2475-4ee5-ab89-8a353371a374.png)

the code is written for a directory structure like the following

![image](https://user-images.githubusercontent.com/62752094/135721243-42363ebd-9ffc-4d66-a7db-a834264f9821.png)

help me 
Thank you

**Comment by MuhammadAsadJaved** [comment]
> > Hi @hanifjunos @jhonjam , You can convert VisDrone annotations to Darknet format by this simple script.
> > https://github.com/zhaobaiyu/visdrone#scripts-for-dataset-transform
> 
> yes, I am trying to use that code. However, it asks as input in sequence set, which does not have it. those are the downloaded files from the dataset (http://aiskyeye.com/download/object-detection-2/) ![image](https://user-images.githubusercontent.com/62752094/135721170-2f6144b2-2475-4ee5-ab89-8a353371a374.png)
> 
> the code is written for a directory structure like the following
> 
> ![image](https://user-images.githubusercontent.com/62752094/135721243-42363ebd-9ffc-4d66-a7db-a834264f9821.png)
> 
> help me Thank you

There are two solutions.
1- You can download the dataset from his given links. 
2- You can change the code a little bit as per your data directories.  How is your dataset? all images in one folder and all annotations in the second folder?

**Comment by mohammed2100** [comment]
Hi, could anyone explain how to convert the labels of visDrone datasets labels to make them suit for YOLOv7.
Thanks a lot.
